### PR TITLE
deploy version 1.2.0

### PR DESCRIPTION
**Pull request by megies**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

## DO NOT MERGE YET -- TESTING BUILD WITH RELEASE CANDIDATES

### CODE REVIEW

**Comment by megies** [comment]
@conda-forge-admin, please rerender

**Comment by megies** [comment]
@conda-forge/core tried to rerender locally, but can't get current smithy installed

<details>
 <summary>Fails: conda create -n conda-smithy python=3.7 conda-smithy</summary>

```
$ conda create -n conda-smithy python=3.7 conda-smithy
Collecting package metadata: done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.6.4
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /home/megies/anaconda/envs/conda-smithy

  added / updated specs:
    - blas[build=openblas]
    - conda-smithy
    - flake8
    - pip
    - python=3.7


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge
    _openmp_mutex-4.5          |            0_gnu         435 KB  conda-forge
    beautifulsoup4-4.8.2       |           py37_0         157 KB  conda-forge
    blas-2.14                  |         openblas          10 KB  conda-forge
    bzip2-1.0.8                |       h516909a_2         396 KB  conda-forge
    certifi-2019.11.28         |           py37_0         148 KB  conda-forge
    cffi-1.13.2                |   py37h8022711_0         220 KB  conda-forge
    conda-4.8.0                |           py37_1         3.0 MB  conda-forge
    conda-build-3.18.11        |           py37_0         520 KB  conda-forge
    conda-forge-pinning-2020.01.24|                0           8 KB  conda-forge
    conda-package-handling-1.6.0|   py37h516909a_1         942 KB  conda-forge
    conda-smithy-3.6.5         |             py_0          58 KB  conda-forge
    cryptography-2.8           |   py37h72c5cf5_1         625 KB  conda-forge
    deprecated-1.2.7           |             py_0          10 KB  conda-forge
    filelock-3.0.10            |             py_0           9 KB  conda-forge
    git-2.24.0                 |  pl526hce37bd2_1        14.6 MB  conda-forge
    gitdb2-2.0.6               |             py_0          46 KB  conda-forge
    gitpython-3.0.5            |             py_0         346 KB  conda-forge
    glob2-0.7                  |             py_0          11 KB  conda-forge
    isodate-0.6.0              |             py_1          25 KB  conda-forge
    jinja2-2.10.3              |             py_0          91 KB  conda-forge
    krb5-1.16.4                |       h2fd8d38_0         1.4 MB  conda-forge
    ld_impl_linux-64-2.33.1    |       h53a641e_7         653 KB  conda-forge
    libarchive-3.3.3           |    hc47fbbf_1007         1.5 MB  conda-forge
    libblas-3.8.0              |      14_openblas          10 KB  conda-forge
    libcblas-3.8.0             |      14_openblas          10 KB  conda-forge
    liblapack-3.8.0            |      14_openblas          10 KB  conda-forge
    liblapacke-3.8.0           |      14_openblas          10 KB  conda-forge
    liblief-0.9.0              |       hf8a498c_1         4.2 MB  conda-forge
    markupsafe-1.1.1           |   py37h516909a_0          26 KB  conda-forge
    msrest-0.6.10              |             py_0          47 KB  conda-forge
    oauthlib-3.0.1             |             py_0          82 KB  conda-forge
    patchelf-0.10              |       he1b5a44_0          77 KB  conda-forge
    perl-5.26.2                |    h516909a_1006        15.4 MB  conda-forge
    pip-20.0.1                 |           py37_0         1.9 MB  conda-forge
    pkginfo-1.5.0.1            |             py_0          21 KB  conda-forge
    psutil-5.6.7               |   py37h516909a_0         323 KB  conda-forge
    py-lief-0.9.0              |   py37he1b5a44_1         1.4 MB  conda-forge
    pycosat-0.6.3              |py37h516909a_1002         105 KB  conda-forge
    pycrypto-2.6.1             |py37h516909a_1003         490 KB  conda-forge
    pygithub-1.43.6            |           py37_0         2.0 MB  conda-forge
    pyjwt-1.7.1                |             py_0          17 KB  conda-forge
    pyopenssl-19.1.0           |           py37_0          83 KB  conda-forge
    pysocks-1.7.1              |           py37_0          27 KB  conda-forge
    python-3.7.6               |       h357f687_2        52.9 MB  conda-forge
    python-libarchive-c-2.9    |           py37_0          54 KB  conda-forge
    pyyaml-5.3                 |   py37h516909a_0         184 KB  conda-forge
    requests-2.22.0            |           py37_1          84 KB  conda-forge
    requests-oauthlib-1.2.0    |             py_0          19 KB  conda-forge
    ripgrep-11.0.2             |       h516909a_3         1.8 MB  conda-forge
    ruamel.yaml-0.16.6         |   py37h516909a_0         166 KB  conda-forge
    ruamel.yaml.clib-0.2.0     |   py37h516909a_0         155 KB  conda-forge
    ruamel_yaml-0.15.80        |py37h516909a_1000         268 KB  conda-forge
    setuptools-45.1.0          |           py37_0         648 KB  conda-forge
    six-1.14.0                 |           py37_0          23 KB  conda-forge
    soupsieve-1.9.4            |           py37_0          57 KB  conda-forge
    tqdm-4.41.1                |             py_0          45 KB  conda-forge
    urllib3-1.25.7             |           py37_0         159 KB  conda-forge
    vsts-python-api-0.1.22     |             py_0         544 KB  conda-forge
    yaml-0.2.2                 |       h516909a_1          82 KB  conda-forge
    zstd-1.4.4                 |       h3b9ef0a_1         989 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       109.4 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-0_gnu
  beautifulsoup4     conda-forge/linux-64::beautifulsoup4-4.8.2-py37_0
  blas               conda-forge/linux-64::blas-2.14-openblas
  blinker            conda-forge/noarch::blinker-1.4-py_1
  bzip2              conda-forge/linux-64::bzip2-1.0.8-h516909a_2
  ca-certificates    conda-forge/linux-64::ca-certificates-2019.11.28-hecc5488_0
  certifi            conda-forge/linux-64::certifi-2019.11.28-py37_0
  cffi               conda-forge/linux-64::cffi-1.13.2-py37h8022711_0
  chardet            conda-forge/linux-64::chardet-3.0.4-py37_1003
  conda              conda-forge/linux-64::conda-4.8.0-py37_1
  conda-build        conda-forge/linux-64::conda-build-3.18.11-py37_0
  conda-forge-pinni~ conda-forge/noarch::conda-forge-pinning-2020.01.24-0
  conda-package-han~ conda-forge/linux-64::conda-package-handling-1.6.0-py37h516909a_1
  conda-smithy       conda-forge/noarch::conda-smithy-3.6.5-py_0
  cryptography       conda-forge/linux-64::cryptography-2.8-py37h72c5cf5_1
  curl               conda-forge/linux-64::curl-7.65.3-hf8cf82a_0
  deprecated         conda-forge/noarch::deprecated-1.2.7-py_0
  entrypoints        conda-forge/linux-64::entrypoints-0.3-py37_1000
  expat              conda-forge/linux-64::expat-2.2.5-he1b5a44_1004
  filelock           conda-forge/noarch::filelock-3.0.10-py_0
  flake8             conda-forge/linux-64::flake8-3.7.9-py37_0
  gettext            conda-forge/linux-64::gettext-0.19.8.1-hc5be6a0_1002
  git                conda-forge/linux-64::git-2.24.0-pl526hce37bd2_1
  gitdb2             conda-forge/noarch::gitdb2-2.0.6-py_0
  gitpython          conda-forge/noarch::gitpython-3.0.5-py_0
  glob2              conda-forge/noarch::glob2-0.7-py_0
  icu                conda-forge/linux-64::icu-64.2-he1b5a44_1
  idna               conda-forge/linux-64::idna-2.8-py37_1000
  isodate            conda-forge/noarch::isodate-0.6.0-py_1
  jinja2             conda-forge/noarch::jinja2-2.10.3-py_0
  krb5               conda-forge/linux-64::krb5-1.16.4-h2fd8d38_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libarchive         conda-forge/linux-64::libarchive-3.3.3-hc47fbbf_1007
  libblas            conda-forge/linux-64::libblas-3.8.0-14_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-14_openblas
  libcurl            conda-forge/linux-64::libcurl-7.65.3-hda55be3_0
  libedit            conda-forge/linux-64::libedit-3.1.20170329-hf8c457e_1001
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1006
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.2.0-h24d8f2e_2
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.3.0-hdf63c60_4
  libgomp            conda-forge/linux-64::libgomp-9.2.0-h24d8f2e_2
  libiconv           conda-forge/linux-64::libiconv-1.15-h516909a_1005
  liblapack          conda-forge/linux-64::liblapack-3.8.0-14_openblas
  liblapacke         conda-forge/linux-64::liblapacke-3.8.0-14_openblas
  liblief            conda-forge/linux-64::liblief-0.9.0-hf8a498c_1
  libopenblas        conda-forge/linux-64::libopenblas-0.3.7-h5ec1e0e_6
  libssh2            conda-forge/linux-64::libssh2-1.8.2-h22169c7_2
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.2.0-hdf63c60_2
  libxml2            conda-forge/linux-64::libxml2-2.9.10-hee79883_0
  lz4-c              conda-forge/linux-64::lz4-c-1.8.3-he1b5a44_1001
  lzo                conda-forge/linux-64::lzo-2.10-h14c3975_1000
  markupsafe         conda-forge/linux-64::markupsafe-1.1.1-py37h516909a_0
  mccabe             conda-forge/noarch::mccabe-0.6.1-py_1
  msrest             conda-forge/noarch::msrest-0.6.10-py_0
  ncurses            conda-forge/linux-64::ncurses-6.1-hf484d3e_1002
  oauthlib           conda-forge/noarch::oauthlib-3.0.1-py_0
  openssl            conda-forge/linux-64::openssl-1.1.1d-h516909a_0
  patchelf           conda-forge/linux-64::patchelf-0.10-he1b5a44_0
  pcre               conda-forge/linux-64::pcre-8.43-he1b5a44_0
  perl               conda-forge/linux-64::perl-5.26.2-h516909a_1006
  pip                conda-forge/linux-64::pip-20.0.1-py37_0
  pkginfo            conda-forge/noarch::pkginfo-1.5.0.1-py_0
  psutil             conda-forge/linux-64::psutil-5.6.7-py37h516909a_0
  py-lief            conda-forge/linux-64::py-lief-0.9.0-py37he1b5a44_1
  pycodestyle        conda-forge/noarch::pycodestyle-2.5.0-py_0
  pycosat            conda-forge/linux-64::pycosat-0.6.3-py37h516909a_1002
  pycparser          conda-forge/linux-64::pycparser-2.19-py37_1
  pycrypto           conda-forge/linux-64::pycrypto-2.6.1-py37h516909a_1003
  pyflakes           conda-forge/noarch::pyflakes-2.1.1-py_0
  pygithub           conda-forge/linux-64::pygithub-1.43.6-py37_0
  pyjwt              conda-forge/noarch::pyjwt-1.7.1-py_0
  pyopenssl          conda-forge/linux-64::pyopenssl-19.1.0-py37_0
  pysocks            conda-forge/linux-64::pysocks-1.7.1-py37_0
  python             conda-forge/linux-64::python-3.7.6-h357f687_2
  python-libarchive~ conda-forge/linux-64::python-libarchive-c-2.9-py37_0
  pytz               conda-forge/noarch::pytz-2019.3-py_0
  pyyaml             conda-forge/linux-64::pyyaml-5.3-py37h516909a_0
  readline           conda-forge/linux-64::readline-8.0-hf8c457e_0
  requests           conda-forge/linux-64::requests-2.22.0-py37_1
  requests-oauthlib  conda-forge/noarch::requests-oauthlib-1.2.0-py_0
  ripgrep            conda-forge/linux-64::ripgrep-11.0.2-h516909a_3
  ruamel.yaml        conda-forge/linux-64::ruamel.yaml-0.16.6-py37h516909a_0
  ruamel.yaml.clib   conda-forge/linux-64::ruamel.yaml.clib-0.2.0-py37h516909a_0
  ruamel_yaml        conda-forge/linux-64::ruamel_yaml-0.15.80-py37h516909a_1000
  setuptools         conda-forge/linux-64::setuptools-45.1.0-py37_0
  six                conda-forge/linux-64::six-1.14.0-py37_0
  smmap2             conda-forge/noarch::smmap2-2.0.5-py_0
  soupsieve          conda-forge/linux-64::soupsieve-1.9.4-py37_0
  sqlite             conda-forge/linux-64::sqlite-3.30.1-hcee41ef_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  toolz              conda-forge/noarch::toolz-0.10.0-py_0
  tqdm               conda-forge/noarch::tqdm-4.41.1-py_0
  urllib3            conda-forge/linux-64::urllib3-1.25.7-py37_0
  vsts-python-api    conda-forge/noarch::vsts-python-api-0.1.22-py_0
  wheel              conda-forge/linux-64::wheel-0.33.6-py37_0
  wrapt              conda-forge/linux-64::wrapt-1.11.2-py37h516909a_0
  xz                 conda-forge/linux-64::xz-5.2.4-h14c3975_1001
  yaml               conda-forge/linux-64::yaml-0.2.2-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1006
  zstd               conda-forge/linux-64::zstd-1.4.4-h3b9ef0a_1


Proceed ([y]/n)? 


Downloading and Extracting Packages
ruamel.yaml.clib-0.2 | 155 KB    | ##################################################################################################################################### | 100% 
urllib3-1.25.7       | 159 KB    | ##################################################################################################################################### | 100% 
ripgrep-11.0.2       | 1.8 MB    | ##################################################################################################################################### | 100% 
psutil-5.6.7         | 323 KB    | ##################################################################################################################################### | 100% 
tqdm-4.41.1          | 45 KB     | ##################################################################################################################################### | 100% 
blas-2.14            | 10 KB     | ##################################################################################################################################### | 100% 
isodate-0.6.0        | 25 KB     | ##################################################################################################################################### | 100% 
oauthlib-3.0.1       | 82 KB     | ##################################################################################################################################### | 100% 
cffi-1.13.2          | 220 KB    | ##################################################################################################################################### | 100% 
ruamel.yaml-0.16.6   | 166 KB    | ##################################################################################################################################### | 100% 
libcblas-3.8.0       | 10 KB     | ##################################################################################################################################### | 100% 
ruamel_yaml-0.15.80  | 268 KB    | ##################################################################################################################################### | 100% 
py-lief-0.9.0        | 1.4 MB    | ##################################################################################################################################### | 100% 
pygithub-1.43.6      | 2.0 MB    | ##################################################################################################################################### | 100% 
libarchive-3.3.3     | 1.5 MB    | ##################################################################################################################################### | 100% 
krb5-1.16.4          | 1.4 MB    | ##################################################################################################################################### | 100% 
pip-20.0.1           | 1.9 MB    | ##################################################################################################################################### | 100% 
conda-4.8.0          | 3.0 MB    | ##################################################################################################################################### | 100% 
perl-5.26.2          | 15.4 MB   | ##################################################################################################################################### | 100% 
yaml-0.2.2           | 82 KB     | ##################################################################################################################################### | 100% 
deprecated-1.2.7     | 10 KB     | ##################################################################################################################################### | 100% 
patchelf-0.10        | 77 KB     | ##################################################################################################################################### | 100% 
pyjwt-1.7.1          | 17 KB     | ##################################################################################################################################### | 100% 
certifi-2019.11.28   | 148 KB    | ##################################################################################################################################### | 100% 
vsts-python-api-0.1. | 544 KB    | ##################################################################################################################################### | 100% 
pycrypto-2.6.1       | 490 KB    | ##################################################################################################################################### | 100% 
python-3.7.6         | 52.9 MB   | ##################################################################################################################################### | 100% 
jinja2-2.10.3        | 91 KB     | ##################################################################################################################################### | 100% 
requests-2.22.0      | 84 KB     | ##################################################################################################################################### | 100% 
zstd-1.4.4           | 989 KB    | ##################################################################################################################################### | 100% 
requests-oauthlib-1. | 19 KB     | ##################################################################################################################################### | 100% 
setuptools-45.1.0    | 648 KB    | ##################################################################################################################################### | 100% 
pycosat-0.6.3        | 105 KB    | ##################################################################################################################################### | 100% 
markupsafe-1.1.1     | 26 KB     | ##################################################################################################################################### | 100% 
six-1.14.0           | 23 KB     | ##################################################################################################################################### | 100% 
liblief-0.9.0        | 4.2 MB    | ##################################################################################################################################### | 100% 
pyopenssl-19.1.0     | 83 KB     | ##################################################################################################################################### | 100% 
conda-build-3.18.11  | 520 KB    | ##################################################################################################################################### | 100% 
git-2.24.0           | 14.6 MB   | ##################################################################################################################################### | 100% 
pyyaml-5.3           | 184 KB    | ##################################################################################################################################### | 100% 
ld_impl_linux-64-2.3 | 653 KB    | ##################################################################################################################################### | 100% 
libblas-3.8.0        | 10 KB     | ##################################################################################################################################### | 100% 
soupsieve-1.9.4      | 57 KB     | ##################################################################################################################################### | 100% 
cryptography-2.8     | 625 KB    | ##################################################################################################################################### | 100% 
pysocks-1.7.1        | 27 KB     | ##################################################################################################################################### | 100% 
glob2-0.7            | 11 KB     | ##################################################################################################################################### | 100% 
msrest-0.6.10        | 47 KB     | ##################################################################################################################################### | 100% 
beautifulsoup4-4.8.2 | 157 KB    | ##################################################################################################################################### | 100% 
conda-forge-pinning- | 8 KB      | ##################################################################################################################################### | 100% 
_libgcc_mutex-0.1    | 3 KB      | ##################################################################################################################################### | 100% 
_openmp_mutex-4.5    | 435 KB    | ##################################################################################################################################### | 100% 
gitdb2-2.0.6         | 46 KB     | ##################################################################################################################################### | 100% 
python-libarchive-c- | 54 KB     | ##################################################################################################################################### | 100% 
gitpython-3.0.5      | 346 KB    | ##################################################################################################################################### | 100% 
pkginfo-1.5.0.1      | 21 KB     | ##################################################################################################################################### | 100% 
liblapacke-3.8.0     | 10 KB     | ##################################################################################################################################### | 100% 
bzip2-1.0.8          | 396 KB    | ##################################################################################################################################### | 100% 
conda-smithy-3.6.5   | 58 KB     | ##################################################################################################################################### | 100% 
liblapack-3.8.0      | 10 KB     | ##################################################################################################################################### | 100% 
filelock-3.0.10      | 9 KB      | ##################################################################################################################################### | 100% 
conda-package-handli | 942 KB    | ##################################################################################################################################### | 100% 

UnicodeDecodeError('ascii', '/info/test/tests/data/\xed\x94\x84\xeb\xa1\x9c\xea\xb7\xb8\xeb\x9e\xa8.README', 22, 23, 'ordinal not in range(128)')
```
</details>

P.S.: I know my conda is not on latest version, but `conda update -n base conda` fills up ~10GB RAM, tries to boil some eggs on my keyboard and does not complete in a reasonable amount of time

**Comment by CJ-Wright** [comment]
I think this needs to point to the RC branch as well.

**Comment by megies** [comment]
> I think this needs to point to the RC branch as well.

I'm sorry, but what do you mean by that?

**Comment by CJ-Wright** [comment]
We very recently adopted a new policy toward dev/rc releases https://github.com/conda-forge/cfep/blob/master/cfep-05.md

We usually do not let rc releases into the master branch (and the main conda-forge channel).

**Comment by megies** [comment]
Oh, I see, thanks for the link. I wasn't planning on deploying any RC builds, though, what I usually did was to see if they CI build and tests are OK and then finally only deploy the actual release.

**Comment by megies** [comment]
@conda-forge-admin, please rerender

**Comment by megies** [comment]
> Did you check the "Allow edits from maintainers" box?

Uhm.. I don't see that box, neither on this open PR nor when opening a new PR.

**Comment by mbargull** [comment]
Next to your first comment on the right you should have a column with:
> - Reviewers
> - Assignees
> - Labels
> - Projects
> - Milestone
> - Notifications
> - n participants
> - Lock conversation
> - [ ] Allow edits from maintainers

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
   fn: obspy-{{ version }}.zip
-  url: https://pypi.io/packages/source/o/obspy/obspy-{{ version }}.zip
-  # url: https://github.com/obspy/obspy/files/2872202/obspy-1.1.1rc4.zip
+  # url: https://pypi.io/packages/source/o/obspy/obspy-{{ version }}.zip
+  url: https://github.com/obspy/obspy/files/4108546/obspy-1.2.0rc1.zip
   sha256: {{ sha256 }}
 
+# for RC builds
+channel_targets:
+  - conda-forge rc_obspy
```
Comment (mbargull) [inline recipe/meta.yaml]: This has to go into `conda_build_config.yaml`.
Also note that `obspy_rc` would be the preferred naming, we have a pending change for that: https://github.com/conda-forge/cfep/pull/3#issuecomment-578364893

**Comment by megies** [comment]
> Next to your first comment on the right you should have a column with:

<details>
 <summary>screenshot</summary>

Screenshot from 2020-01-29 13-51-36
</details>

for new PR:

<details>
 <summary>screenshot</summary>

Screenshot from 2020-01-29 12-56-12
</details>

**Comment by mbargull** [comment]
Ah, I just checked: For PRs from my personal profile (`mbargull`) I do have that option, for PRs from an "organization" (`mbargull-feedstocks`) I don't.
You could either reopen a PR from a repository owned by your account or simply rerender locally and push those changes.
(And I'll be looking for if there are some option on the "organization" repos that would allow edits...)

**Comment by megies** [comment]
Nice catch @mbargull. In any case, I made a fresh miniconda installation just for rerendering, since in my existing one, I can't update conda anymore it does not finish and installing smithy in a new environment crashes (presumably due to a conda bug that was fixed in a newer version).

**Comment by megies** [comment]
Rebased and force pushed without the RC label, we don't have a need to deploy the RC build really, and having the test reports sent from CI to our test overview pages at obspy is sufficient. We will only merge this with the build of the final obspy 1.2.0 release when it is ready.

**Comment by megies** [comment]
The py27 CI fails with a weird error, apparently the installed executable scripts get executed with a wrong Python interpreter (py3)


https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/115616/logs/85
```
2020-01-29T13:48:00.7340240Z 
2020-01-29T13:48:00.7342150Z ----------------------------------------------------------------------
2020-01-29T13:48:00.7342840Z Ran 1958 tests in 396.035s
2020-01-29T13:48:00.7343400Z 
2020-01-29T13:48:00.7344100Z OK
2020-01-29T13:48:11.7747180Z 
2020-01-29T13:48:12.9635240Z Your test results have been reported and are available at: http://tests.obspy.org/108341/
2020-01-29T13:48:12.9637180Z Thank you!
2020-01-29T13:48:12.9673380Z ===== obspy-1.2.0rc1-py27h3b54f70_0 OK =====
2020-01-29T13:48:12.9674740Z import: 'obspy'
2020-01-29T13:48:12.9675840Z import: 'obspy.io.mseed'
2020-01-29T13:48:13.2736660Z + python -c 'from obspy import read;'
2020-01-29T13:48:14.5099450Z + obspy-print -h
2020-01-29T13:48:14.5342280Z   File "/usr/local/miniconda/conda-bld/obspy_1580304851728/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/obspy-print", line 5
2020-01-29T13:48:14.5350940Z     from obspy.scripts.print import main
2020-01-29T13:48:14.5351610Z                            ^
2020-01-29T13:48:14.5352030Z SyntaxError: invalid syntax
2020-01-29T13:48:16.4711650Z Tests failed for obspy-1.2.0rc1-py27h3b54f70_0.tar.bz2 - moving package to /usr/local/miniconda/conda-bld/broken
```

Is there a way to access the built packages even if they are not deployed (access as artifact on CI job?)?

**Comment by mbargull** [comment]
> Is there a way to access the built packages even if they are not deployed (access as artifact on CI job?)?

No, I don't believe there is (sadly).

> The py27 CI fails with a weird error, apparently the installed executable scripts get executed with a wrong Python interpreter (py3)

Nope, if it were executed with Py3 it wouldn't error out ;). This is a legitimate Py2-related error. The entry point scripts that are created do not `from __future__ import print_function` beforehand and thus it somehow trips over the `print` "keyword" in the `from obspy.scripts.print import main` line.

**Comment by megies** [comment]
> Nope, if it were executed with Py3 it wouldn't error out ;). This is a legitimate Py2-related error.

Oh.. good point. I managed to reproduce it locally, so hopefully I can work around it somehow.

**Comment by mbargull** [comment]
Probably the easiest thing (apart from dropping 2.7 support right away :P) would be to just create an indirection via something like `obspy/scripts/_print.py`:
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Print stream information for waveform data in local files.
"""
from __future__ import (absolute_import, division, print_function,
                        unicode_literals)

from .print import main

main = main
```
and set the entry point to `obspy.scripts._print:main`.

(Another alternative would be, of course, to not use `pip install` to get the old `setuptools`/`pkg_resources` entry points back, but I don't like to promote their usage in any case ;)

**Comment by mbargull** [comment]
There is another failure at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=115616&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=d66ed8ca-07ff-5419-c5d1-5e21fa500bb4&l=4890:
```
b'<title>SDS Report [\\w:/\\\\]*?[/\\\\]obspy-sdstest-[\\w]*</title>'
b'<title>SDS Report C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\obspy-sdstest-n71c856h</title>'
b'SDS Report for [\\w:/\\\\]*?[/\\\\]obspy-sdstest-[\\w]*<br>'
b'SDS Report for C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\obspy-sdstest-n71c856h<br>'
```
judging by the looks of it, this comes from https://github.com/obspy/obspy/blob/1.2.0rc1/obspy/clients/filesystem/tests/test_sds.py#L214-L225 and might be due to the `/`s that are expected by https://github.com/obspy/obspy/blob/1.2.0rc1/obspy/clients/filesystem/tests/data/sds_report.regex#L5.

**Comment by megies** [comment]
Thanks a lot for the tip @mbargull, that indirection is indeed probably the easiest solution.

**Comment by mbargull** [comment]
Oh, apologies, I've misread before and thought the `/` was outside of the brackets..

But the solution is even easier than that: The path has been shortened to ye old 8.3 scheme with à la `UPPERC~1.EXT`. You just have to add `~` to the allowed characters, i.e., change https://github.com/obspy/obspy/blob/1.2.0rc1/obspy/clients/filesystem/tests/data/sds_report.regex#L5 like so:
```diff
-<title>SDS Report [\w:/\\]*?[/\\]obspy-sdstest-[\w]*</title>
+<title>SDS Report [\w:/\\~]*?[/\\]obspy-sdstest-[\w]*</title>
```

**Comment by megies** [comment]
Thanks @mbargull, I played around locally with adding a tilde but with all the escaping going on I couldn't confirm it was just the `~`, so I went catch all with `.*` since this is really just a minor point in the test anyway and I wanted to avoid more iterations through RCs and CI.

**Comment by megies** [comment]
@ThomasLecocq any idea regarding the [failing console script calls](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=115906&view=logs&j=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&t=0d1d4714-3221-52cd-cfdb-f62a891e036b&l=7357)?

Wait.. Windows doesn't use dash for command line options but slashes, right?
So is it `obspy-print /h` to get the help printed on Windows?

CC @barsch 

```
Ran 1958 tests in 233.234s

OK

Your test results have been reported and are available at: http://tests.obspy.org/108382/
Thank you!
===== obspy-1.2.0rc3-py36hc8d92b1_0 OK =====
import: 'obspy'
import: 'obspy.io.mseed'

(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit 1 

(%PREFIX%) %SRC_DIR%>call "%SRC_DIR%\run_test.bat" 

(%PREFIX%) %SRC_DIR%>python -c "from obspy import read;" 

(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit /B 1 

(%PREFIX%) %SRC_DIR%>obspy-print -h 

(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit /B 1 

(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit 1 
Tests failed for obspy-1.2.0rc3-py36hc8d92b1_0.tar.bz2 - moving package to D:\bld\broken
```

v1.1.1 build on Windows also had problems calling those scripts but still passed somehow btw.. https://ci.appveyor.com/project/conda-forge/obspy-feedstock/builds/26476698/job/xkj5d3xqksss7j8j#L6219

**Comment by megies** [comment]
OK build looks good. **Do not merge** will update with final release when it's ready.

File: recipe/meta.yaml (hunk 1 above)
Comment (jasongrout) [reply recipe/meta.yaml]: (see also https://github.com/conda-forge/cfep/pull/21 reflecting the preferred naming)

Comment (megies) [reply recipe/meta.yaml]: Thanks! However, we don't actually intend to merge (and thus deploy) the RCs, I was just making sure the builds look good in CI and we do not have packaging problems. We're expecting to push out the release in the next few days and I'll update this PR with the release zipball.

**Comment by ThomasLecocq** [comment]
@megies anything I can do / help here ?

**Comment by megies** [comment]
My bad, didn't update the static file link for the RC build here @ThomasLecocq

**Pull request merged**